lib/rex/socket.rb

```diff
@@ -186,14 +186,12 @@ def self.getaddresses(hostname, accept_ipv6 = true)
       return [hostname]
     end
 
-    res = ::Socket.gethostbyname(hostname)
-    return [] if not res
+    res = ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
 
-    # Shift the first three elements out, leaving just the list of
-    # addresses
-    res.shift # name
-    res.shift # alias hostnames
-    res.shift # address_family
+    res.map! do |address_info|
+      address_info.ip_address
+    end
+    return [] if not res
```

Comment:
This seems like an implementation hangover that could probably be deleted once the below comment is resolved

Reply:
I went for this as before it was shifting the other information out of the array and only returning the IP address. 

I just replicated that, unless you're referring to removing line 191-194 so we return the full address info and not just the IP addresses. 

Whats returned without those lines:
```Ruby
[#<Addrinfo: 127.0.0.1 TCP (localhost)>, #<Addrinfo: ::1 TCP (localhost)>]
```

With those lines:
```Ruby
["127.0.0.1", "::1"]
```

Maybe I'm misunderstanding what you mean though?

Reply:
> I think these tests need updated similar to how .getaddresses was - rather than deleted 👀

Got those tests added back in and updated.